capi-matrix/README.md

```diff
@@ -0,0 +1,78 @@
+
+- As Giant Swarm staff, I would like to release a new OS version with the same Kubernetes version and roll all customer clusters when a security breach is detected.
+
+## Solution
+
+The solution proposed is to create a cluster-PROVIDER chart release for every Kubernetes/OS version is released. That way will help our automation to roll upgrades automatically when new version is published.
+
+But we would like to have a list of images attached to every Kubernetes version like:
+
+```yaml
+kubernetesVersionImages:
+  v1.22.9: "ubuntu-2004-kube-v1.22.9"
+  v1.22.10: "ubuntu-2004-kube-v1.22.10"
+  v1.23.3: "ubuntu-2004-kube-v1.23.3"
+  v1.23.4: "ubuntu-2004-kube-v1.23.4"
+  v1.23.5: "ubuntu-2004-kube-v1.23.5"
+```
+
+So the customers don't have to think about image names and we can remove Kubernetes/OS versions that are flawed. For example we can change OS version keeping the Kubernetes version the same:
+
+```yaml
+kubernetesVersionImages:
+  v1.22.9: "ubuntu-2005-kube-v1.22.9"
+  ...
+```
+
+Every time there is a new Kubernetes version or OS version we will (automatically) create a PR to cluster-PROVIDER charts including the version and potentially changing the default version (if our tests run successfully).
+
+At the same time if we found a Kubernetes/OS version with a security issue, we can remove the Kubernetes/OS version and enforce customers to upgrade clusters.
+
+We have start with [Cluster API OpenStack](https://github.com/giantswarm/cluster-openstack/pull/125).
+
+__Bonus__: In favour of making the providers apps not tied to Giant Swarm, adding the conversion image table helps enabling other vendors to adopt it and customize it.
+
+## How the conversion image table is updated?
+
+- We have not defined this process. But one possible scenario would be:
+
+1) Kubernetes/Kinvolk release a new OS/k8s version
+
+2) Our image builder job is triggered and creates the AMIs for the different providers. 
+
+3) Other job can be triggered after and create pull requests to the cluster-PROVIDER repos adding the new k8s/OS version.
+
+4) Our future CI pipeline can test this new version for each provider and check it runs correctly (conformance tests or even custom tests). 
+
+5) In case CI succeed the default version and conversion image table is changed to the new one with the approval of the owning team.
+
+From this moment Cluster Upgrade automation can start scheduling the workload cluster upgrades based on customer constraints. More info about this process discussed [here](https://github.com/giantswarm/rfc/pull/52).
+
+## Open questions
+
+- Every time we release a new provider app with a newer Kubernetes version we can make it default, but at the same time we might want to wait and run not edge version but a previous one. So we will need a bit of process deciding when we change the default version.
+
+## Out of scope
+
+- How containerd version is managed.
```

Comment:
if i understand correctly, we would manage containerd versions with this rfc, as they're tied to os versions

Reply:
then yes, but I meant here don't want to bring containerd version to the mix. I assume the containerd version will be compatible with the k8s components running on the OS

Reply:
But our idea is to moved Flatcar in all providers, right?